sr/robot/arenas/caldera_arena.py

```diff
@@ -0,0 +1,125 @@
+    return (x * cos(radians) + y * -sin(radians)), (x * sin(radians) + y * cos(radians))
+
+
+PLATFORM_COLOUR = (0x68, 0x75, 0x91)
+
+class CalderaArena(Arena):
+    start_locations = [(-3.6, -3.6),
+                       (3.6, -3.6),
+                       (3.6, 3.6),
+                       (-3.6, 3.6)]
+
+    start_headings = [0.25 * pi,
+                      0.75 * pi,
+                      -0.75 * pi,
+                      -0.25 * pi]
+
+    def __init__(self, objects=None, wall_markers=True):
+        super(CalderaArena, self).__init__(objects, wall_markers)
+        self._init_tokens()
+
+    def _init_tokens(self):
+        token_ids = [28, 29, 30, 31]
+
+        token_locations_offsets_from_zone = [(0.5, 1.3),
+                                             (0.5, 1.5),
+                                             (0.5, 1.7),
+                                             (0.5, 1.9),
+                                             (0.5, 2.1),]
+
+        for zone in range(0, 4):
+            for i, location in enumerate(token_locations_offsets_from_zone):
+                rotated_location = rotate(location[0] - 4, location[1] - 4, (pi / 2) * zone)
+                token = Token(self, token_ids[zone], damping=0.5)
+                token.location = rotated_location
+                self.objects.append(token)
+
+    def draw_background(self, surface, display):
+        super(CalderaArena, self).draw_background(surface, display)
+
+        def line(start, end):
+            pygame.draw.line(surface, ARENA_MARKINGS_COLOR,
+                             display.to_pixel_coord(start), display.to_pixel_coord(end),
+                             ARENA_MARKINGS_WIDTH)
+
+        def line_symmetric(start, end):
+            """
+            Draw a line, double reflected on the X and Y axis
+            (creating 8 lines)
+            """
+            start_x, start_y = start
+            end_x, end_y = end
+            line((start_x, start_y), (end_x, end_y))
+            line((-start_x, start_y), (-end_x, end_y))
+            line((-start_x, -start_y), (-end_x, -end_y))
+            line((start_x, -start_y), (end_x, -end_y))
+            line((start_y, start_x), (end_y, end_x))
+            line((-start_y, start_x), (-end_y, end_x))
+            line((-start_y, -start_x), (-end_y, -end_x))
+            line((start_y, -start_x), (end_y, -end_x))
+
+        # Starting zones
+        self.starting_zone_side = 1
+        draw_corner_zones(self, display, surface, shape='Square')
+
+
+        # Grid
+        square_width = 1.200
+        start_pos = (-3, -3)
+        grid_dimensions = (5, 5)
+        quartered_grid_dims = (int(ceil(grid_dimensions[0]/2)), int(ceil(grid_dimensions[1]/2)))
+
+        def to_grid_pos(grid_x, grid_y):
+            return (start_pos[0] + grid_x * square_width,
+                    start_pos[1] + grid_y * square_width)
+
+        # Raised platform
+        # (These positions are in grid cell locations)
```

Comment:
I think it's fairly important that we make the platform something which the robots can't "just" drive over.
I suggest having a look at the way that the Tin Can Rally simulator works for how to do walls inside the arena.

It seems reasonable to me that we make the simulator robot not able to climb on the platform given the other limitations that the simulated robot has.

A more advanced thing to do might be to have a command-line switch which flattens the platform, or possibly simulator API call (`R.climb()`?) which allows the robot to climb the platform.